src/test/issues.c2.test.ts

```diff
@@ -91,6 +92,21 @@ describe("Issue Tests", () => {
         );
     });
 
+    it("#2574", () => {
+        const project = convert();
+        const usesDefaultExport = query(project, "usesDefaultExport");
+        const sig = usesDefaultExport.signatures?.[0];
+        ok(sig, "Missing signature for usesDefaultExport");
+        const param = sig.parameters?.[0];
+        ok(param, "Missing parameter");
+        equal(param.name, "param", "Incorrect parameter name");
+        const paramType = param.type as ReferenceType | undefined;
+        ok(paramType, "Parameter type is not a reference type or undefined");
+        equal(paramType.type, "reference", "Parameter is not a reference type");
+        equal(paramType.name, "DefaultExport", "Incorrect reference name");
```

Comment:
I was expecting to see `Default` here, `DefaultExport` is definitely not the right thing to show to the user as that name is internal to the module where the symbol is declared, `Default` also matches what TypeScript shows when hovering over the type.

Also, please stick this test at the end of the file, it's intended to be in order of the issue number so it's easy to find tests

Reply:
> I was expecting to see Default here, DefaultExport is definitely not the right thing to show to the user as that name is internal to the module where the symbol is declared, Default also matches what TypeScript shows when hovering over the type.

See, the weird thing is that TypeScript's `checker.getTypeFromSymbolAtLocation()` function returns the underlying `DefaultExport` in this instance, so I figured that's just how it was expected to show up. I'll see what I can do to get it to behave correctly, but I was having a lot of trouble getting that to work.

Reply:
> Default also matches what TypeScript shows when hovering over the type.

When hovering over the type, it does show `Default` as the name and export. When navigating to that type, the LSP seems to bring me to the `DefaultExport` implementation, not the `export { default as Default }` statement at the file that is actually imported. It seems that TypeScript's behavior is set to resolve all references deeply to the base type implementation rather than showing the reference type from the re-export.

Reply:
Hmm... I guess that result is reasonable then

Reply:
Think I should log this as an issue on [microsoft/TypeScript](https://github.com/microsoft/TypeScript)?